src/main/java/org/gridsuite/securityanalysis/server/repositories/ContingencyRepository.java

```diff
@@ -7,17 +7,62 @@
 
 package org.gridsuite.securityanalysis.server.repositories;
 
+import jakarta.persistence.criteria.*;
+import org.gridsuite.securityanalysis.server.dto.ResourceFilterDTO;
 import org.gridsuite.securityanalysis.server.entities.ContingencyEntity;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 import java.util.UUID;
+
 /**
  * @author Kevin Le Saulnier <kevin.lesaulnier at rte-france.com>
  */
 
 @Repository
-public interface ContingencyRepository extends JpaRepository<ContingencyEntity, UUID> {
-    List<ContingencyEntity> findByResultIdAndStatusOrderByContingencyId(UUID resultUuid, String status);
+public interface ContingencyRepository extends CommonLimitViolationRepository<ContingencyEntity>, JpaRepository<ContingencyEntity, UUID>, JpaSpecificationExecutor<ContingencyEntity> {
+    Page<ContingencyEntity> findAll(Specification<ContingencyEntity> specification, Pageable pageable);
+
+    default void addPredicate(CriteriaBuilder criteriaBuilder,
+                                      Root<ContingencyEntity> path,
+                                      List<Predicate> predicates,
+                                      ResourceFilterDTO filter) {
+
+        String fieldName = switch (filter.column()) {
+            case CONTINGENCY_ID -> "contingencyId";
+            case STATUS -> "status";
+            default -> throw new UnsupportedOperationException("This method should be called for parent filters only");
+        };
+
+        CriteriaUtils.addPredicate(criteriaBuilder, path, predicates, filter, fieldName);
+    }
+
+    default void addJoinFilter(CriteriaBuilder criteriaBuilder,
+                                      Join<?, ?> joinPath,
+                                      ResourceFilterDTO filter) {
+        String fieldName;
+        String subFieldName = null;
+
+        switch (filter.column()) {
+            case SUBJECT_ID -> {
```

Comment:
Instead of doing all these switch, I would add a field inside the enum, like `columnPath` of something like that.
Furthermore, I don't think using fieldName, subFieldName, etc... Is a good method. What if you want to access to a subSub or subSubSub field, etc ?
You can copy paste and use this method if you want : https://github.com/gridsuite/shortcircuit-server/blob/31bd1f1b475b9ea53a4a508692773dbc0563a037/src/main/java/org/gridsuite/shortcircuit/server/utils/SpecificationUtils.java#L124-L135

Reply:
I would not add anything to the enum since the column path depends on the repository we're using

fieldName/subFieldName code has been taken from sensitivity-analysis-server, but alright I will make it better

Reply:
removed field/subfield but kept switch since each repository can have a different way to reach fields